scripts/vm.sh

```
@@ -56,16 +56,123 @@ function create_vms() {
+    generate_mac_with_custom_prefix() {
+        local vm_index="$1"
+        local custom_prefix="$2"
+        
+        # Validate custom prefix format (should be 2 hex digits or 4 hex digits with colon)
+        if [[ ! "$custom_prefix" =~ ^[0-9A-Fa-f]{2}$ ]] && [[ ! "$custom_prefix" =~ ^[0-9A-Fa-f]{2}:[0-9A-Fa-f]{2}$ ]]; then
+            log "ERROR" "Invalid MAC_CUSTOM_PREFIX format: $custom_prefix. Must be 2 hexadecimal digits (e.g., '01', 'A1') or 4 hex digits with colon (e.g., 'C0:00')"
+            exit 1
+        fi
+        
+        # Convert VM index to hex (01, 02, 03, etc.)
+        local last_octet_hex=$(printf '%02x' "$vm_index")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)